src/configuration/AzureAuthSettings.tsx

```diff
@@ -15,23 +15,21 @@
 //
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
-
-import React, { FunctionComponent, FormEvent, useMemo, useState } from 'react';
+import { cx } from '@emotion/css';
+import React, { FormEvent, useMemo, useState } from 'react';
 
 import { config } from '@grafana/runtime';
 import { InlineField, InlineFieldRow, InlineSwitch, Input } from '@grafana/ui';
-import { HttpSettingsBaseProps } from '@grafana/ui/components/DataSourceSettings/types';
+
+import { HttpSettingsBaseProps } from "../components/types";
 
 import { KnownAzureClouds, AzureCredentials } from './AzureCredentials';
 import { getCredentials, updateCredentials } from './AzureCredentialsConfig';
 import { AzureCredentialsForm } from './AzureCredentialsForm';
 
-export const AzureAuthSettings: FunctionComponent<HttpSettingsBaseProps> = (props: HttpSettingsBaseProps) => {
+export const AzureAuthSettings = (props: HttpSettingsBaseProps) => {
   const { dataSourceConfig, onChange } = props;
 
-  const [overrideAudienceAllowed] = useState<boolean>(
-    config.featureToggles.prometheusAzureOverrideAudience || !!dataSourceConfig.jsonData.azureEndpointResourceId
-  );
```

Comment:
we do not need this state?

Reply:
Now `disabled={dataSourceConfig.readOnly}` is used